### PR TITLE
Remove deprecated methods/params into Cart

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove deprecated in Cart
| Type?             | improvement 
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #26293.
| How to test?      | Nothing to test.
| Possible impacts? | BC Break.

**:notebook: BC Breaks**
* Removed const : `Cart::ONLY_PRODUCTS_WITHOUT_SHIPPING`
* Removed method : `Cart::getTaxesAverageUsed()`
* Removed method : `Cart::containsProduct()`
* Removed method : `Cart::simulateCarriersOutput()`
* Removed method : `Cart::getOrderShippingCost()`
* Removed method : `Cart::checkDiscountValidity()`
* Removed method : `Cart::deletePictureToProduct()`
* Removed method : `Cart::addExtraCarriers()`
* Removed parameters : `$exclusive`, in `Cart::isAllProductsInStock()`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26302)

### CODE REVIEW

**Review by tegbessou — APPROVED** [review]
Thank you for your PR, nice work! Seems to be good IMO

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@PrestaEdit Could you fix CI and PHPStan errors ?

File: classes/Cart.php [hunk 1]
```diff
@@ -5049,20 +4878,13 @@ public function checkAllProductsAreStillAvailableInThisState()
...
      *
      * @param bool $ignore_virtual Ignore virtual products
-     * @param bool $exclusive (DEPRECATED) If true, the validation is exclusive : it must be present product in stock and out of stock
      *
      * @since 1.5.0
      *
      * @return bool False if not all products in the cart are in stock
      */
-    public function isAllProductsInStock($ignoreVirtual = false, $exclusive = false)
+    public function isAllProductsInStock($ignoreVirtual = false)
```
Comment (matthieu-rolland) [inline classes/Cart.php:4886]: you remove this parameter but the variable is still used in the function

below there's a condition, `if (!$exclusive)` followed by `elseif ($exclusive)`, if this parameter must really disappear you need to remove these conditions and define what is the default behavior

**Review by matthieu-rolland — COMMENTED** [review]
one feedback

File: classes/Cart.php (hunk 1 above)
Comment (PrestaEdit) [reply classes/Cart.php:4886]: Good catch, don't know why I'm missing it.

Mmmmmh. As the default is "false" and it's deprecated, wdyt ? Do we need to keep only the false part, maybe ?

Comment (matthieu-rolland) [reply classes/Cart.php:4886]: yes, if we remove this optional parameter, we snould keep the default behavior, which is `$eclusive = false` 👍

**Comment by PierreRambaud** [comment]
Hi @PrestaEdit ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**